--- /openshift-ansible/roles/openshift_facts/library/openshift_facts.py	2018-06-06 18:26:01.150426798 +0000
+++ /openshift-ansible/roles/openshift_facts/library/openshift_facts.py.new	2018-06-06 15:06:20.000000000 +0000
@@ -639,6 +639,9 @@ def set_nodename(facts):
         if 'cloudprovider' in facts and facts['cloudprovider']['kind'] == 'gce':
             facts['node']['nodename'] = facts['provider']['metadata']['instance']['hostname'].split('.')[0]

+        elif 'cloudprovider' in facts and facts['cloudprovider']['kind'] == 'azure':
+            facts['node']['nodename'] = facts['common']['hostname'].lower().split('.')[0]
+
         # TODO: The openstack cloudprovider nodename setting was too opinionaed.
         #       It needs to be generalized before it can be enabled again.
         # elif 'cloudprovider' in facts and facts['cloudprovider']['kind'] == 'openstack':
